In [1]:
!pip install apyori

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Created wheel for apyori: filename=apyori-1.1.2-py3-none-any.whl size=5974 sha256=823fc4c70aa068a8b5c5d745ed46d623e48ba2c9b84a6de33898bff3579a64ac
  Stored in directory: /root/.cache/pip/wheels/cb/f6/e1/57973c631d27efd1a2f375bd6a83b2a616c4021f24aab84080
Successfully built apyori


In [2]:
import numpy as np
import pandas as pd
from apyori import apriori

In [3]:
store_data = pd.read_csv('https://raw.githubusercontent.com/HuynhHoa99/AprioriAlgorithm/master/Day1.csv', header=None)


In [4]:
store_data.head()


0      1      2       3     4      5
0  Wine  Chips  Bread  Butter  Milk  Apple
1  Wine    NaN  Bread  Butter  Milk    NaN
2   NaN    NaN  Bread  Butter  Milk    NaN
3   NaN  Chips    NaN     NaN   NaN  Apple
4  Wine  Chips  Bread  Butter  Milk  Apple

In [5]:
store_data.shape


(22, 6)

In [6]:
records = []
for i in range(22):
    records.append([str(store_data.values[i, j]) for j in range(6)])

In [8]:
records

[['Wine', 'Chips', 'Bread', 'Butter', 'Milk', 'Apple'],
 ['Wine', 'nan', 'Bread', 'Butter', 'Milk', 'nan'],
 ['nan', 'nan', 'Bread', 'Butter', 'Milk', 'nan'],
 ['nan', 'Chips', 'nan', 'nan', 'nan', 'Apple'],
 ['Wine', 'Chips', 'Bread', 'Butter', 'Milk', 'Apple'],
 ['Wine', 'Chips', 'nan', 'nan', 'Milk', 'nan'],
 ['Wine', 'Chips', 'Bread', 'Butter', 'nan', 'Apple'],
 ['Wine', 'Chips', 'nan', 'nan', 'Milk', 'nan'],
 ['Wine', 'nan', 'Bread', 'nan', 'nan', 'Apple'],
 ['Wine', 'nan', 'Bread', 'Butter', 'Milk', 'nan'],
 ['nan', 'Chips', 'Bread', 'Butter', 'nan', 'Apple'],
 ['Wine', 'nan', 'nan', 'Butter', 'Milk', 'Apple'],
 ['Wine', 'Chips', 'Bread', 'Butter', 'Milk', 'nan'],
 ['Wine', 'nan', 'Bread', 'nan', 'Milk', 'Apple'],
 ['Wine', 'nan', 'Bread', 'Butter', 'Milk', 'Apple'],
 ['Wine', 'Chips', 'Bread', 'Butter', 'Milk', 'Apple'],
 ['nan', 'Chips', 'Bread', 'Butter', 'Milk', 'Apple'],
 ['nan', 'Chips', 'nan', 'Butter', 'Milk', 'Apple'],
 ['Wine', 'Chips', 'Bread', 'Butter', 'Milk', 'Apple

In [10]:
association_rules = apriori(records, min_support=0.5, min_confidence=0.7, min_lift=1.2, min_length=2)

association_results = list(association_rules)
association_results

[RelationRecord(items=frozenset({'Butter', 'Bread', 'Milk'}), support=0.5, ordered_statistics=[OrderedStatistic(items_base=frozenset({'Butter'}), items_add=frozenset({'Bread', 'Milk'}), confidence=0.7333333333333334, lift=1.241025641025641), OrderedStatistic(items_base=frozenset({'Bread', 'Milk'}), items_add=frozenset({'Butter'}), confidence=0.8461538461538461, lift=1.241025641025641)])]

In [ ]:
print(association_results)


[RelationRecord(items=frozenset({'Bread', 'Butter', 'Milk'}), support=0.5, ordered_statistics=[OrderedStatistic(items_base=frozenset({'Butter'}), items_add=frozenset({'Bread', 'Milk'}), confidence=0.7333333333333334, lift=1.241025641025641), OrderedStatistic(items_base=frozenset({'Bread', 'Milk'}), items_add=frozenset({'Butter'}), confidence=0.8461538461538461, lift=1.241025641025641)])]


=> Quy luật được phát hiện là những giao dịch mà mua Bread và Milk thì 84.6% cũng mua Butter , và 1 lần Butter được mua không thì sẽ có 1.241 lần Butter sẽ được mua kèm với những người mua bread và milk


Practice with groceries dataset

In [12]:
import numpy as np
import pandas as pd
from mlxtend.frequent_patterns import apriori, association_rules

In [13]:
data = pd.read_excel('http://archive.ics.uci.edu/ml/machine-learning-databases/00352/Online%20Retail.xlsx')

In [14]:
data.head()

InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

          InvoiceDate  UnitPrice  CustomerID         Country  
0 2010-12-01 08:26:00       2.55     17850.0  United Kingdom  
1 2010-12-01 08:26:00       3.39     17850.0  United Kingdom  
2 2010-12-01 08:26:00       2.75     17850.0  United Kingdom  
3 2010-12-01 08:26:00       3.39     17850.0  United Kingdom  
4 2010-12-01 08:26:00       3.39     17850.0  United Kingdom

In [15]:
data.columns

Index(['InvoiceNo', 'StockCode', 'Description', 'Quantity', 'InvoiceDate',
       'UnitPrice', 'CustomerID', 'Country'],
      dtype='object')

In [16]:
data.Country.unique()

array(['United Kingdom', 'France', 'Australia', 'Netherlands', 'Germany',
       'Norway', 'EIRE', 'Switzerland', 'Spain', 'Poland', 'Portugal',
       'Italy', 'Belgium', 'Lithuania', 'Japan', 'Iceland',
       'Channel Islands', 'Denmark', 'Cyprus', 'Sweden', 'Austria',
       'Israel', 'Finland', 'Bahrain', 'Greece', 'Hong Kong', 'Singapore',
       'Lebanon', 'United Arab Emirates', 'Saudi Arabia',
       'Czech Republic', 'Canada', 'Unspecified', 'Brazil', 'USA',
       'European Community', 'Malta', 'RSA'], dtype=object)

In [17]:
data.describe()

Quantity      UnitPrice     CustomerID
count  541909.000000  541909.000000  406829.000000
mean        9.552250       4.611114   15287.690570
std       218.081158      96.759853    1713.600303
min    -80995.000000  -11062.060000   12346.000000
25%         1.000000       1.250000   13953.000000
50%         3.000000       2.080000   15152.000000
75%        10.000000       4.130000   16791.000000
max     80995.000000   38970.000000   18287.000000

In [18]:
data['Description']

0          WHITE HANGING HEART T-LIGHT HOLDER
1                         WHITE METAL LANTERN
2              CREAM CUPID HEARTS COAT HANGER
3         KNITTED UNION FLAG HOT WATER BOTTLE
4              RED WOOLLY HOTTIE WHITE HEART.
                         ...                 
541904            PACK OF 20 SPACEBOY NAPKINS
541905           CHILDREN'S APRON DOLLY GIRL 
541906          CHILDRENS CUTLERY DOLLY GIRL 
541907        CHILDRENS CUTLERY CIRCUS PARADE
541908          BAKING SET 9 PIECE RETROSPOT 
Name: Description, Length: 541909, dtype: object

In [19]:
data['Description'] = data['Description'].str.strip()
data['Description']

0          WHITE HANGING HEART T-LIGHT HOLDER
1                         WHITE METAL LANTERN
2              CREAM CUPID HEARTS COAT HANGER
3         KNITTED UNION FLAG HOT WATER BOTTLE
4              RED WOOLLY HOTTIE WHITE HEART.
                         ...                 
541904            PACK OF 20 SPACEBOY NAPKINS
541905            CHILDREN'S APRON DOLLY GIRL
541906           CHILDRENS CUTLERY DOLLY GIRL
541907        CHILDRENS CUTLERY CIRCUS PARADE
541908           BAKING SET 9 PIECE RETROSPOT
Name: Description, Length: 541909, dtype: object

In [24]:
data.dropna(axis = 0, subset =['InvoiceNo'], inplace = True)
data['InvoiceNo'] = data['InvoiceNo'].astype('str')

In [25]:
data.dtypes

InvoiceNo              object
StockCode              object
Description            object
Quantity                int64
InvoiceDate    datetime64[ns]
UnitPrice             float64
CustomerID            float64
Country                object
dtype: object

In [28]:
~data['InvoiceNo'].str.contains('C')

0         True
1         True
2         True
3         True
4         True
          ... 
541904    True
541905    True
541906    True
541907    True
541908    True
Name: InvoiceNo, Length: 541909, dtype: bool

In [29]:
data = data[~data['InvoiceNo'].str.contains('C')]

In [35]:
sum_data_france = data[data['Country'] =="France"].groupby(['InvoiceNo', 'Description'])['Quantity'].sum()
sum_data_france

InvoiceNo  Description                    
536370     ALARM CLOCK BAKELIKE GREEN         12
           ALARM CLOCK BAKELIKE PINK          24
           ALARM CLOCK BAKELIKE RED           24
           CHARLOTTE BAG DOLLY GIRL DESIGN    20
           CIRCUS PARADE LUNCH BOX            24
                                              ..
581587     CIRCUS PARADE LUNCH BOX            12
           PACK OF 20 SPACEBOY NAPKINS        12
           PLASTERS IN TIN CIRCUS PARADE      12
           PLASTERS IN TIN STRONGMAN          12
           SPACEBOY LUNCH BOX                 12
Name: Quantity, Length: 8372, dtype: int64

In [44]:
basket_data_france = (data[data['Country'] =="France"].groupby(['InvoiceNo', 'Description'])['Quantity'].sum().unstack().reset_index().fillna(0).set_index('InvoiceNo'))
basket_data_france.head()

Description  10 COLOUR SPACEBOY PEN  12 COLOURED PARTY BALLOONS  \
InvoiceNo                                                         
536370                          0.0                         0.0   
536852                          0.0                         0.0   
536974                          0.0                         0.0   
537065                          0.0                         0.0   
537463                          0.0                         0.0   

Description  12 EGG HOUSE PAINTED WOOD  12 MESSAGE CARDS WITH ENVELOPES  \
InvoiceNo                                                                 
536370                             0.0                              0.0   
536852                             0.0                              0.0   
536974                             0.0                              0.0   
537065                             0.0                              0.0   
537463                             0.0                              0.0   

Description  12 PENCIL SMALL TUBE WOODLAND  \
InvoiceNo                                    
536370                                 0.0   
536852                                 0.0   
536974                                 0.0   
537065                                 0.0   
537463                                 0.0   

Description  12 PENCILS SMALL TUBE RED RETROSPOT  12 PENCILS SMALL TUBE SKULL  \
InvoiceNo                                                                       
536370                                       0.0                          0.0   
536852                                       0.0                          0.0   
536974                                       0.0                          0.0   
537065                                       0.0                          0.0   
537463                                       0.0                          0.0   

Description  12 PENCILS TALL TUBE POSY  12 PENCILS TALL TUBE RED RETROSPOT  \
InvoiceNo                                                                    
536370                             0.0                                 0.0   
536852                             0.0                                 0.0   
536974                             0.0                                 0.0   
537065                             0.0                                 0.0   
537463                             0.0                                 0.0   

Description  12 PENCILS TALL TUBE WOODLAND  ...  WRAP VINTAGE PETALS  DESIGN  \
InvoiceNo                                   ...                                
536370                                 0.0  ...                          0.0   
536852                                 0.0  ...                          0.0   
536974                                 0.0  ...                          0.0   
537065                                 0.0  ...                          0.0   
537463                                 0.0  ...                          0.0   

Description  YELLOW COAT RACK PARIS FASHION  YELLOW GIANT GARDEN THERMOMETER  \
InvoiceNo                                                                      
536370                                  0.0                              0.0   
536852                                  0.0                              0.0   
536974                                  0.0                              0.0   
537065                                  0.0                              0.0   
537463                                  0.0                              0.0   

Description  YELLOW SHARK HELICOPTER  ZINC  STAR T-LIGHT HOLDER  \
InvoiceNo                                                         
536370                           0.0                        0.0   
536852                           0.0                        0.0   
536974                           0.0                        0.0   
537065                           0.0                        0.0   
537463                           0.0      

In [45]:
# Transactions done in France
basket_France = (data[data['Country'] =="France"]
          .groupby(['InvoiceNo', 'Description'])['Quantity']
          .sum().unstack().reset_index().fillna(0)
          .set_index('InvoiceNo'))

In [46]:
# Transactions done in the United Kingdom
basket_UK = (data[data['Country'] =="United Kingdom"]
          .groupby(['InvoiceNo', 'Description'])['Quantity']
          .sum().unstack().reset_index().fillna(0)
          .set_index('InvoiceNo'))


In [57]:
sum_basket_Por = data[data['Country'] =="Portugal"].groupby(['InvoiceNo', 'Description'])['Quantity'].sum()
sum_basket_Por.head()

InvoiceNo  Description           
536990     LUNCH BAG CARS BLUE       14
           LUNCH BAG SUKI  DESIGN    10
           LUNCH BAG WOODLAND        10
           POSTAGE                    1
           RED HEART LUGGAGE TAG     12
Name: Quantity, dtype: int64

In [54]:
# Transactions done in Portugal
basket_Por = (data[data['Country'] =="Portugal"]
          .groupby(['InvoiceNo', 'Description'])['Quantity']
          .sum().unstack().reset_index().fillna(0)
          .set_index('InvoiceNo'))
basket_Por


Description  10 COLOUR SPACEBOY PEN  12 PENCIL SMALL TUBE WOODLAND  \
InvoiceNo                                                            
536990                          0.0                            0.0   
537246                          0.0                            0.0   
537818                          0.0                            0.0   
537915                          0.0                            0.0   
538311                          0.0                            0.0   
539353                          0.0                            0.0   
540519                          0.0                            0.0   
540546                         24.0                            0.0   
541430                          0.0                            0.0   
542147                          0.0                            0.0   
544495                          0.0                            0.0   
545191                          0.0                            0.0   
545937                          0.0                            0.0   
547444                          0.0                            0.0   
547566                          0.0                            0.0   
547897                          0.0                            0.0   
548470                          0.0                            0.0   
549435                          0.0                            0.0   
549953                          0.0                            0.0   
553017                          0.0                            2.0   
553018                          0.0                            0.0   
554155                          0.0                            0.0   
554301                          0.0                            0.0   
554305                          0.0                            0.0   
555862                          0.0                            0.0   
556024                          0.0                            0.0   
556829                          0.0                            0.0   
559862                          0.0                            0.0   
559907                          0.0                            0.0   
560694                          0.0                            0.0   
563241                          0.0                            0.0   
565295                          0.0                            0.0   
567929                          0.0                            0.0   
568208                          0.0                            0.0   
569866                          0.0                            3.0   
569870                          0.0                            0.0   
570499                          0.0                            0.0   
570964                          0.0                            0.0   
571188                          0.0                            0.0   
571324                          0.0                            0.0   
571326                          0.0                            0.0   
571477                          0.0                            0.0   
571901                          0.0                            0.0   
572680                          0.0                            0.0   
574099                          0.0                            0.0   
574844                          0.0                            0.0   
575743                          0.0                            0.0   
576257                          0.0                            0.0   
576269                          0.0                            0.0   
576895                          0.0                            0.0   
577618                          0.0                            0.0   
578463                          0.0                            0.0   
578488                          0.0                            0.0   
579790                          0.0                            0.0   
580153                          0.0                            0.0   
580545    

In [ ]:
basket_Por['']

In [48]:
basket_Sweden = (data[data['Country'] =="Sweden"]
          .groupby(['InvoiceNo', 'Description'])['Quantity']
          .sum().unstack().reset_index().fillna(0)
          .set_index('InvoiceNo'))

In [49]:
def hot_encode(x):
    if(x<= 0):
        return 0
    if(x>= 1):
        return 1

In [50]:
# Encoding the datasets
basket_encoded = basket_France.applymap(hot_encode)
basket_France = basket_encoded
basket_France

Description  10 COLOUR SPACEBOY PEN  12 COLOURED PARTY BALLOONS  \
InvoiceNo                                                         
536370                            0                           0   
536852                            0                           0   
536974                            0                           0   
537065                            0                           0   
537463                            0                           0   
...                             ...                         ...   
580986                            0                           0   
581001                            0                           0   
581171                            0                           0   
581279                            0                           0   
581587                            0                           0   

Description  12 EGG HOUSE PAINTED WOOD  12 MESSAGE CARDS WITH ENVELOPES  \
InvoiceNo                                                                 
536370                               0                                0   
536852                               0                                0   
536974                               0                                0   
537065                               0                                0   
537463                               0                                0   
...                                ...                              ...   
580986                               0                                0   
581001                               0                                0   
581171                               0                                0   
581279                               0                                0   
581587                               0                                0   

Description  12 PENCIL SMALL TUBE WOODLAND  \
InvoiceNo                                    
536370                                   0   
536852                                   0   
536974                                   0   
537065                                   0   
537463                                   0   
...                                    ...   
580986                                   0   
581001                                   0   
581171                                   0   
581279                                   0   
581587                                   0   

Description  12 PENCILS SMALL TUBE RED RETROSPOT  12 PENCILS SMALL TUBE SKULL  \
InvoiceNo                                                                       
536370                                         0                            0   
536852                                         0                            0   
536974                                         0                            0   
537065                                         0                            0   
537463                                         0                            0   
...                                          ...                          ...   
580986                                         0                            0   
581001                                         0                            0   
581171                                         0                            0   
581279                                         0                            0   
581587                                         0                            0   

Description  12 PENCILS TALL TUBE POSY  12 PENCILS TALL TUBE RED RETROSPOT  \
InvoiceNo                                                                    
536370                               0                                   0   
536852                               0                                   0   
536974                               0                                   0   
537065                               0                                   0   
537463                         

In [58]:
# Encoding the datasets
basket_encoded = basket_France.applymap(hot_encode)
basket_France = basket_encoded
 
basket_encoded = basket_UK.applymap(hot_encode)
basket_UK = basket_encoded
 
basket_encoded = basket_Por.applymap(hot_encode)
basket_Por = basket_encoded
 
basket_encoded = basket_Sweden.applymap(hot_encode)
basket_Sweden = basket_encoded

In [60]:
# Building the model
frq_items = apriori(basket_France, min_support = 0.05, use_colnames = True)
frq_items

support                                           itemsets
0    0.071429                      (4 TRADITIONAL SPINNING TOPS)
1    0.096939                       (ALARM CLOCK BAKELIKE GREEN)
2    0.102041                        (ALARM CLOCK BAKELIKE PINK)
3    0.094388                         (ALARM CLOCK BAKELIKE RED)
4    0.068878                       (ASSORTED COLOUR MINI CASES)
..        ...                                                ...
190  0.102041  (SET/6 RED SPOTTY PAPER CUPS, POSTAGE, SET/6 R...
191  0.099490  (SET/6 RED SPOTTY PAPER CUPS, SET/20 RED RETRO...
192  0.056122  (POSTAGE, ALARM CLOCK BAKELIKE GREEN, ALARM CL...
193  0.053571  (PLASTERS IN TIN SPACEBOY, PLASTERS IN TIN CIR...
194  0.081633  (SET/6 RED SPOTTY PAPER CUPS, SET/20 RED RETRO...

[195 rows x 2 columns]

France results

In [61]:
# Collecting the inferred rules in a dataframe
rules = association_rules(frq_items, metric ="lift", min_threshold = 1)
rules = rules.sort_values(['confidence', 'lift'], ascending =[False, False])
print(rules.head())

                                           antecedents  \
44                        (JUMBO BAG WOODLAND ANIMALS)   
260  (PLASTERS IN TIN CIRCUS PARADE, RED TOADSTOOL ...   
272  (RED TOADSTOOL LED NIGHT LIGHT, PLASTERS IN TI...   
300  (SET/6 RED SPOTTY PAPER CUPS, SET/20 RED RETRO...   
302  (SET/6 RED SPOTTY PAPER PLATES, SET/20 RED RET...   

                         consequents  antecedent support  consequent support  \
44                         (POSTAGE)            0.076531            0.765306   
260                        (POSTAGE)            0.051020            0.765306   
272                        (POSTAGE)            0.053571            0.765306   
300  (SET/6 RED SPOTTY PAPER PLATES)            0.102041            0.127551   
302    (SET/6 RED SPOTTY PAPER CUPS)            0.102041            0.137755   

      support  confidence      lift  leverage  conviction  
44   0.076531       1.000  1.306667  0.017961         inf  
260  0.051020       1.000  1.306667  0.011974     

UK results

In [62]:
frq_items = apriori(basket_UK, min_support = 0.01, use_colnames = True)
rules = association_rules(frq_items, metric ="lift", min_threshold = 1)
rules = rules.sort_values(['confidence', 'lift'], ascending =[False, False])
print(rules.head())

KeyboardInterrupt: ignored

Portugal results

In [ ]:
frq_items = apriori(basket_Por, min_support = 0.05, use_colnames = True)
rules = association_rules(frq_items, metric ="lift", min_threshold = 1)
rules = rules.sort_values(['confidence', 'lift'], ascending =[False, False])
print(rules.head())

Sweden 

In [ ]:

frq_items = apriori(basket_Sweden, min_support = 0.05, use_colnames = True)
rules = association_rules(frq_items, metric ="lift", min_threshold = 1)
rules = rules.sort_values(['confidence', 'lift'], ascending =[False, False])
print(rules.head())

Nhận xét kết quả

Eclat 

In [63]:
!pip install pyECLAT


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [64]:
transactions = [
    ['beer', 'wine', 'cheese'],
    ['beer', 'potato chips'],
    ['eggs', 'flower', 'butter', 'cheese'],
    ['eggs', 'flower', 'butter', 'beer', 'potato chips'],
    ['wine', 'cheese'],
    ['potato chips'],
    ['eggs', 'flower', 'butter', 'wine', 'cheese'],
    ['eggs', 'flower', 'butter', 'beer', 'potato chips'],
    ['wine', 'beer'],
    ['beer', 'potato chips'],
    ['butter', 'eggs'],
    ['beer', 'potato chips'],
    ['flower', 'eggs'],
    ['beer', 'potato chips'],
    ['eggs', 'flower', 'butter', 'wine', 'cheese'],
    ['beer', 'wine', 'potato chips', 'cheese'],
    ['wine', 'cheese'],
    ['beer', 'potato chips'],
    ['wine', 'cheese'],
    ['beer', 'potato chips']
]

In [65]:
transactions

[['beer', 'wine', 'cheese'],
 ['beer', 'potato chips'],
 ['eggs', 'flower', 'butter', 'cheese'],
 ['eggs', 'flower', 'butter', 'beer', 'potato chips'],
 ['wine', 'cheese'],
 ['potato chips'],
 ['eggs', 'flower', 'butter', 'wine', 'cheese'],
 ['eggs', 'flower', 'butter', 'beer', 'potato chips'],
 ['wine', 'beer'],
 ['beer', 'potato chips'],
 ['butter', 'eggs'],
 ['beer', 'potato chips'],
 ['flower', 'eggs'],
 ['beer', 'potato chips'],
 ['eggs', 'flower', 'butter', 'wine', 'cheese'],
 ['beer', 'wine', 'potato chips', 'cheese'],
 ['wine', 'cheese'],
 ['beer', 'potato chips'],
 ['wine', 'cheese'],
 ['beer', 'potato chips']]

In [66]:
import pandas as pd
data = pd.DataFrame(transactions)
data

0             1             2       3             4
0           beer          wine        cheese    None          None
1           beer  potato chips          None    None          None
2           eggs        flower        butter  cheese          None
3           eggs        flower        butter    beer  potato chips
4           wine        cheese          None    None          None
5   potato chips          None          None    None          None
6           eggs        flower        butter    wine        cheese
7           eggs        flower        butter    beer  potato chips
8           wine          beer          None    None          None
9           beer  potato chips          None    None          None
10        butter          eggs          None    None          None
11          beer  potato chips          None    None          None
12        flower          eggs          None    None          None
13          beer  potato chips          None    None          None
14          eggs        flower        butter    wine        cheese
15          beer          wine  potato chips  cheese          None
16          wine        cheese          None    None          None
17          beer  potato chips          None    None          None
18          wine        cheese          None    None          None
19          beer  potato chips          None    None          None

In [67]:
min_n_products = 2
min_support = 7/len(transactions)
max_length = max([len(x) for x in transactions])


In [68]:
from pyECLAT import ECLAT


In [69]:
my_eclat = ECLAT(data=data, verbose=True)


100%|██████████| 8/8 [00:00<00:00, 1474.08it/s]


In [70]:
rule_indices, rule_supports = my_eclat.fit(min_support=min_support,
                                           min_combination=min_n_products,
                                           max_combination=max_length)

Combination 2 by 2


10it [00:00, 216.77it/s]


Combination 3 by 3


10it [00:00, 204.91it/s]


Combination 4 by 4


5it [00:00, 195.85it/s]


Combination 5 by 5


1it [00:00, 129.25it/s]


In [71]:
print(rule_supports)


{'beer & potato chips': 0.45, 'wine & cheese': 0.35}
